contracts/mockContracts/MockDGXDex.sol

```diff
@@ -43,91 +33,102 @@ contract MockDGXDEX is Withdrawable {
+        if (blockCheck) {
+            require((blockNumber + BLOCK_EXPIRATION) >= block.number);
         }
 
-        require((block_number + 5) >= block.number);
+        amountMg = (msg.value / weiPerDgxMg);
+        amountMgDgxWei = amountMg * DGX_DECIMALS;
+        amountDgxWei = amountMgDgxWei / GRAM_TO_MILLIGRAM;
 
-        tokenAmount = (msg.value / weiPerDgxMg) * ( DGX_DECIMALS / G_TO_MG);
-        require(token.transfer(msg.sender, tokenAmount));
+        require(token.transfer(msg.sender, amountDgxWei));
 
         success = true;
-        purchasedAmount = tokenAmount;
-        
-        Purchase(success, purchasedAmount);
+        purchasedAmount = amountDgxWei;
+
+        Purchase(success, purchasedAmount, msg.sender, weiPerDgxMg);
         
     }
 
+    event Sell(bool success, uint256 amountWei, address sellerAddress, uint256 weiPerDgxMg);
+
     function sell(uint256 amount,
-                  uint256 block_number,
-                  uint256  nonce,
+                  uint256 blockNumber,
+                  uint256 nonce,
                   uint256 weiPerDgxMg,
                   address signer,
-                  bytes signature) public returns (bool success) {
+                  bytes signature)
+    public
+    returns (bool success) {
 
         uint256 amountWei;
+        uint256 amountDgxWei;
+        uint256 amountMgDgxWei;
+        bool verified;
+        address actualSigner;
 
         if (sigCheck) {
-            bool verified;
-            address actual_signer;
-            
-            (verified, actual_signer) =   
-            verify_signed_price(block_number,
-                                nonce,
-                                weiPerDgxMg,
-                                signer,
-                                signature);
-
+            (verified, actualSigner) = verify_signed_price(blockNumber,
+                                                           nonce,
+                                                           weiPerDgxMg,
+                                                           signer,
+                                                           signature);
             require(verified);
         }
         if (blockCheck) {
-            require((block_number + 5) >= block.number);
+            require((blockNumber + BLOCK_EXPIRATION) >= block.number);
         }
 
-        amountWei = amount * G_TO_MG * weiPerDgxMg / DGX_DECIMALS;
+        amountDgxWei = amount;
+        amountMgDgxWei = amountDgxWei * GRAM_TO_MILLIGRAM;
+        amountWei = amountMgDgxWei * weiPerDgxMg / DGX_DECIMALS;
+
         require(token.transferFrom(msg.sender, this, amount));
         msg.sender.transfer(amountWei);
         
         success = true;
 
-        Sell(success, amountWei);
+        Sell(success, amountWei, msg.sender, weiPerDgxMg);
     }
 
     function verify_signed_price(uint _block_number, uint _nonce, uint _price, address _signer, bytes _signature)
```

Comment:
wrong notation.

Reply:
this is code that we copied from Digix contract.
So I don't want to fix it.
It's good to have it distingushed in our code and I added the comment:
 /* The code below is DGX signature verification (logic and code style). */